mobile/src/main/java/org/openhab/habdroid/background/BackgroundTasksManager.kt

```diff
@@ -145,6 +143,8 @@ class BackgroundTasksManager : BroadcastReceiver() {
                 .build()
             val workRequest = OneTimeWorkRequest.Builder(ItemUpdateWorker::class.java)
                 .setConstraints(constraints)
+                .setBackoffCriteria(BackoffPolicy.EXPONENTIAL,
+                    WorkRequest.MIN_BACKOFF_MILLIS, TimeUnit.MILLISECONDS)
```

Comment:
Isn't this the default anyway? https://developer.android.com/reference/kotlin/androidx/work/WorkRequest.Builder.html#setBackoffCriteria(androidx.work.BackoffPolicy,%20java.time.Duration)

I wonder whether there is an actual use case for exponential backoff in our case? Shouldn't linear backoff not work just fine?

Reply:
The default value is 30 seconds, `MIN_BACKOFF_MILLIS` is 10.

> I wonder whether there is an actual use case for exponential backoff in our case? Shouldn't linear backoff not work just fine?

I want to avoid spamming users that don't have a cloud instance set up. Imagine the alarm clock changes and they're not at home: After 10 * 10s = 100s they get an error notification.
With `EXPONENTIAL` the error is shown much later, but at the same time the first few retries are quite fast.

Reply:
Ah, right. Maybe use different criteria then? Linear for NFC uploads, exponential for background ones.

Reply:
Makes sense.

Reply:
Done